community/modules/scripts/spack-install/variables.tf

```diff
@@ -178,11 +190,32 @@ EOT
 
 variable "environments" {
   description = "Defines a spack environment to configure."
-  default     = null
-  type = list(object({
-    name     = string
-    packages = list(string)
-  }))
+  default     = []
+  type        = list(map(any))
+  validation {
+    condition = alltrue([
+      for e in var.environments : (contains(keys(e), "name") && contains(keys(e), "type"))
+    ])
+    error_message = "All environments must have a name and type."
+  }
+  validation {
+    condition = alltrue([
+      for e in var.environments : (e["type"] == "file" || e["type"] == "packages")
+    ])
+    error_message = "Valid environment types are 'file' and 'packages'."
+  }
+  validation {
+    condition = alltrue([
+      for e in var.environments : ((e["type"] == "packages" && contains(keys(e), "packages")) || e["type"] == "file")
+    ])
+    error_message = "Environments of type 'packages' require a packages list."
+  }
+  validation {
+    condition = alltrue([
+      for e in var.environments : (e["type"] == "packages" || (e["type"] == "file" && contains(keys(e), "value")))
```

Comment:
To keep this consistent with startup scripts, value should be replaced with "content"

Reply:
Thanks, I changed this, and how it's defined in the configs, to help align them with startup scripts.